### DISCUSSION TITLE
How to Display Widget with Data from Relation Manager Table on View Page in Filament?

### DISCUSSION THREAD

**Discussion by Pudge321** [filamentphp/filament#13775] [Help]
### Package

Widgets

### Package Version

v3.2.95

### How can we help you?

I have a relation manager set up, and I want to create a widget that displays specific data from this relation manager on a view page. I’m not sure how to properly fetch and display this data within the widget.

**Comment by leandrocfe** [comment]
Not sure if it works in relation manager
https://filamentphp.com/docs/3.x/panels/resources/widgets#accessing-page-table-data-in-the-widget

**Reply by Pudge321** [reply]
Thank you for your response.

I have tried following the guide, but I keep getting the error: "Property [$activeTab] not found on component" when using use ExposesTableToWidgets; on the view page.

Perhaps this is because the usage differs between the view page and the list page. According to the documentation, it seems that for the list page, I should use use InteractsWithPageTable; on the widget and ExposesTableToWidgets; on the list page.

Meanwhile, for the view and edit pages, I am using public ?Model $record = null;.

Do you have any suggestions or solutions to resolve this?

Thank you.

**Reply by leandrocfe** [reply]
What kind of data do you need to access in the widget?

**Reply by Pudge321** [reply]
I need to show the count of records from the table in the widget, but only for a specific date range using the created_from and until filters in the relation manager. This way, I can display the total number of records created within that time period.

**Reply by aledbrown** [reply]
I've been struggling with this as well but I think I found a solution that will work. You can pass the $record (Model) into the Widget on the View page as a Livewire value, then read it from the public variable on the Widget. Then create stats about it or even talk to relationships - specially the one you are viewing in the Relationship Manager. 

Call the widget in the header or footer of the view page with make and pass the record in.
Widget::make([
                'record' => $this->record,
            ]

Then:
public Model $record; on the Widget.

This was the key info:
https://filamentphp.com/docs/3.x/panels/resources/widgets#passing-properties-to-widgets-on-resource-pages
